tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -383,14 +426,20 @@ static void test_read_abort_callback(const struct device *dev,
 	switch (evt->type) {
 	case UART_TX_DONE:
 		k_sem_give(&tx_done);
+		tx_done_num++;
+		if (tx_done_num == 2) {
+			err = uart_rx_disable(uart_dev);
+			failed_in_isr |= (err < 0);
+		}
```

Comment:
Please see my comments below about changes to this test

Reply:
Reverted